### PR TITLE
[flutter_markdown] Allow for custom block element

### PR DESCRIPTION
**Pull request by dawidope**
Fixes problem with adding custom block syntax.

Issue: https://github.com/flutter/flutter/issues/135848

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter_markdown/lib/src/builder.dart [hunk 1]
```diff
@@ -13,7 +13,7 @@ import '_functions_io.dart' if (dart.library.html) '_functions_web.dart';
 import 'style_sheet.dart';
 import 'widget.dart';
 
-const List<String> _kBlockTags = <String>[
```
Comment (domesticmouse) [inline packages/flutter_markdown/lib/src/builder.dart:16]: Why drop the `const`?

Comment (dawidope) [reply packages/flutter_markdown/lib/src/builder.dart:16]: Because of adding additional tags the list cannot be const anymore.

**Review by domesticmouse — APPROVED** [review]

**Comment by dawidope** [comment]
@domesticmouse is there any blocker to merge it?

**Comment by domesticmouse** [comment]
PTAL @stuartmorgan

**Review by tarrinneal — APPROVED** [review]
updated my requests and merged upstream changes

**Comment by mrares** [comment]
@dawidope I'm very interested in this feature as well and I have a larger patch that also deals with nested blocks and inheriting styles from parent blocks, I'm interested in collaborating with you to upstream it if you're up for it.

**Pull request merged**